File: handlerclaw/lib/features/notifications/data/mappers/notification_mapper.dart
import 'package:handlerclaw/features/notifications/domain/entities/notification_entity.dart';
import 'package:handlerclaw/features/notifications/data/dto/notification_dto.dart';

class NotificationMapper {
  static NotificationEntity fromDto(NotificationDto dto) {
    return NotificationEntity(
      id: dto.id,
      notificationWebhookId: dto.notificationWebhookId,
      userId: dto.userId,
      userDeviceId: dto.userDeviceId,
      title: dto.title,
      body: dto.body,
      data: dto.data,
      triggeredAt: dto.triggeredAt,
      status: dto.status,
      fcmMessageId: dto.fcmMessageId,
      sentAt: dto.sentAt,
      failedAt: dto.failedAt,
      retryCount: dto.retryCount,
      errorMessage: dto.errorMessage,
      createdAt: dto.createdAt,
      updatedAt: dto.updatedAt,
    );
  }

  static List<NotificationEntity> fromDtos(List<NotificationDto> dtos) {
    return dtos.map((dto) => fromDto(dto)).toList();
  }
}

File: api/src/modules/_shared/notifications/infrastructure/mappers/notification.mapper.ts
import type { Notification } from "../../../../../lib/generated/prisma/client.js";
import type { INotification } from "../../domain/entities/notification.entity.js";

export function toNotificationEntity(data: Notification): INotification {
  return {
    id: data.id,
    notification_webhook_id: data.notificationWebhookId,
    user_id: data.userId,
    user_device_id: data.userDeviceId,
    title: data.title,
    body: data.body,
    data: data.data,
    triggered_at: data.triggeredAt,
    status: data.status,
    fcm_message_id: data.fcmMessageId,
    sent_at: data.sentAt,
    failed_at: data.failedAt,
    retry_count: data.retryCount,
    error_message: data.errorMessage,
    created_at: data.createdAt,
    updated_at: data.updatedAt,
  };
}
